src/blueprints/tv2/helpers/tv2-show-style-blueprint-configuration-mapper.ts

```diff
@@ -81,14 +87,32 @@ export class Tv2ShowStyleBlueprintConfigurationMapper {
       graphicsSchemas: this.mapGraphicsSchemas(coreConfiguration.GfxSchemaTemplates),
       selectedGraphicsSetup: this.findSelectedGraphicsSetup(coreConfiguration.GfxDefaults, coreConfiguration.GfxSetups),
       splitScreenConfigurations: this.mapSplitScreenConfigurations(coreConfiguration.DVEStyles),
-      breakerTransitionEffectConfigurations: this.mapTransitionEffectConfigurations([
-        ...coreConfiguration.Transitions.map(transition => transition.Transition),
-        coreConfiguration.ShowstyleTransition
-      ]),
+      // SOF-2004 - Need to handle undefined values from Sofie core
+      // breakerTransitionEffectConfigurations: this.mapTransitionEffectConfigurations([
+      //   ...coreConfiguration.Transitions.map(transition => transition.Transition),
+      //   coreConfiguration.ShowstyleTransition
+      // ]),
+      breakerTransitionEffectConfigurations: this.mapBreakerTransitionEffectConfigurations(
+        coreConfiguration.Transitions, 
+        coreConfiguration.ShowstyleTransition),
       breakers: this.mapToBreakers(coreConfiguration.BreakerConfig)
     }
   }
 
+  private mapBreakerTransitionEffectConfigurations(coreTransitions: Transition[], showstyleTransition: string) : BreakerTransitionEffect[] {
+    //SOF-2004 insert guard clause to protect against undefined values
+    if (!coreTransitions) {
+      return []
+    }
```

Comment:
There is a small inconsistency between the described types and the way they are used.
The type for `coreTransitions` states that it is always an array. So from the type-systems perspective, the expression `!coreTransitions` will never evaluate to true, since an empty array is also a truthy value in JavaScript land :) When working with TypeScript, there is a greater emphasis on the difference between runtime and compile-time data, than perhaps other languages.

From the comment, I assume that the guard is to add the default/fallback value as described in the task. In that case, I would alter the type to reflect that the value can be undefined. By doing that, the types model the real world/runtime data, ensuring that the programmers and the program can make more informed decisions on how to handle said data. It also makes the types tell what the comment currently describes, making the comment redundant. The same goes for the other entries :)

Reply:
I debated with the guys whether to use coreTransitions: Transition[] | undefined or coreTransitions?: Transition[]. 
After some debate we agreed upon the latter approach